core/block_validator.go

```diff
@@ -163,6 +169,67 @@ func (v *BlockValidator) SanityCheckWorkObjectBlockViewBody(wo *types.WorkObject
 	return nil
 }
 
+func (v *BlockValidator) ApplyPoWFilter(wo *types.WorkObject) pubsub.ValidationResult {
+	var err error
+	powhash, exists := v.hc.powHashCache.Peek(wo.Hash())
+	if !exists {
+		powhash, err = v.engine.VerifySeal(wo.WorkObjectHeader())
+		if err != nil {
+			return pubsub.ValidationReject
+		}
+		v.hc.powHashCache.Add(wo.Hash(), powhash)
+	}
+	newBlockIntrinsic := v.engine.IntrinsicLogS(powhash)
+
+	// cannot have a pow filter when the current header is genesis
+	if v.hc.IsGenesisHash(v.hc.CurrentHeader().Hash()) {
+		return pubsub.ValidationAccept
+	}
+
+	currentHeaderPowHash, exists := v.hc.powHashCache.Peek(v.hc.CurrentHeader().Hash())
+	if !exists {
+		currentHeaderPowHash, err = v.engine.VerifySeal(v.hc.CurrentHeader().WorkObjectHeader())
+		if err != nil {
+			return pubsub.ValidationReject
+		}
+		v.hc.powHashCache.Add(v.hc.CurrentHeader().Hash(), currentHeaderPowHash)
+	}
+	currentHeaderIntrinsic := v.engine.IntrinsicLogS(currentHeaderPowHash)
+
+	// Check if the Block is atleast half the current difficulty in Zone Context,
+	// this makes sure that the nodes don't listen to the forks with the PowHash
+	//	with less than 50% of current difficulty
+	if v.hc.NodeCtx() == common.ZONE_CTX && newBlockIntrinsic.Cmp(new(big.Int).Div(currentHeaderIntrinsic, big.NewInt(2))) < 0 {
```

Comment:
Edge case: 7 is less than half of 15, but 15 / 2 = 7 so the block will pass. Probably unlikely

Reply:
yeah agreed, doesnt matter for this case